### PR TITLE
Categorize deprecated warning

### PR DESCRIPTION
**Pull request by nobu**
Warning about deprecation should be categorized as DEPRECATED.

### CODE REVIEW

**Comment by byroot** [comment]
I wonder why it fails on macOS.

File: ext/json/ext/parser/parser.rl [hunk 1]
```diff
@@ -554,7 +563,7 @@ static char *JSON_parse_object(JSON_Parser *json, char *p, char *pe, VALUE *resu
                 VALUE klass = rb_funcall(mJSON, i_deep_const_get, 1, klassname);
                 if (RTEST(rb_funcall(klass, i_json_creatable_p, 0))) {
                     if (json->deprecated_create_additions) {
-                        rb_warn("JSON.load implicit support for `create_additions: true` is deprecated and will be removed in 3.0, use JSON.unsafe_load or explicitly pass `create_additions: true`");
+                        rb_raise(rb_eArgError, deprecated_create_additions_warning);
```
Comment (byroot) [inline ext/json/ext/parser/parser.rl:566]: Ah, did you mean to raise??

Comment (nobu) [reply ext/json/ext/parser/parser.rl]: Seems I pushed wrong commit.

File: ext/json/ext/parser/extconf.rb [hunk 2]
```diff
@@ -5,6 +5,7 @@
 have_func("rb_hash_new_capa", "ruby.h") # RUBY_VERSION >= 3.2
 have_func("rb_gc_mark_locations", "ruby.h") # Missing on TruffleRuby
 have_func("rb_hash_bulk_insert", "ruby.h") # Missing on TruffleRuby
+have_func("rb_category_warn", "ruby.h") # Missing on TruffleRuby
```
Comment (eregon) [inline ext/json/ext/parser/extconf.rb:8]: https://github.com/oracle/truffleruby/issues/3710

**Pull request merged**